docs/UsingSeal.md

```diff
@@ -214,82 +221,84 @@ To decrypt an encrypted object in a Move package, follow these steps:
+// 2. Get derived keys from key servers for this object ID. 
+const derivedKeys = await client.getDerivedKeys({
   id: encryptedObject.id,
-  txBytes, // Should contain call to seal_approve as discussed in the [decryption](#Decryption) section
+  txBytes,
   sessionKey,
   threshold: encryptedObject.threshold,
 });
 
-// Parse encrypted object as Move object
-// For some applications, this object should be stored on-chain (e.g., encrypted votes should be stored before the deadline, and decrypted after).
-const tx = new Transaction();
-const parsedEncryptedObject = tx.moveCall({
-  target: `${SEAL_PACKAGE_ID}::bf_hmac_encryption::parse_encrypted_object`,
-  arguments: [tx.pure.vector("u8", Array.from(encryptedObject.))],
-});
-
-
-// Convert the derived keys to G1 elements
-const derivedKeysAsG1Elements = Array.from(derivedKeys).map(([derivedKey]) => 
-  tx.moveCall({
-    target: `0x2::bls12381::g1_from_bytes`,
-    arguments: [tx.pure.vector("u8", fromHex(derivedKey.toString()))],
-  }),
+// 3. Get the public keys corresponding to the derived keys.
+const publicKeys = await client.getPublicKeys(encryptedObject.services.map(([service, _]) => service));
+const correspondingPublicKeys = derivedKeys.keys().map((objectId) => {
+  const index = encryptedObject.services.findIndex(([s, _]) => s === objectId);
+  return tx.moveCall({
+    target: `${seal_package_id}::bf_hmac_encryption::new_public_key`,
+    arguments: [
+      tx.pure.address(objectId),
+      tx.pure.vector("u8", publicKeys[index].toBytes())
+    ],
+  });
+}).toArray();
+
+// 4. Convert the derived keys to G1 elements.
+const derivedKeysAsG1Elements = Array.from(derivedKeys).map(([_, value]) =>
+tx.moveCall({
+  target: `0x2::bls12381::g1_from_bytes`,
+  arguments: [ tx.pure.vector("u8", fromHex(value.toString())) ],
+})
 );
 
-// Verify the derived keys. This can be cached if decryption for the same ID is done again
+// 5. Call the Move function verify_derived_keys. This should be cached if decryption for the same ID is performed again. 
 const verifiedDerivedKeys = tx.moveCall({
-  target: `${SEAL_PACKAGE_ID}::bf_hmac_encryption::verify_derived_keys`,
+  target: `${seal_package_id}::bf_hmac_encryption::verify_derived_keys`,
   arguments: [
-    tx.makeMoveVec({ elements: derivedKeysAsG1Elements }),
-    tx.pure.address(encryptedObject.packageId),
-    tx.pure.vector("u8", fromHex(encryptedObject.id)),
-    tx.makeMoveVec({ elements: correspondingPublicKeys }),
+  tx.makeMoveVec({ elements: derivedKeysAsG1Elements, type: '0x2::group_ops::Element<0x2::bls12381::G1>' }),
+  tx.pure.address(encryptedObject.packageId),
+  tx.pure.vector("u8", fromHex(encryptedObject.id)),
+    tx.makeMoveVec({ elements: correspondingPublicKeys, type: `${SEAL_PACKAGE_ID}::bf_hmac_encryption::PublicKey` }),
   ],
 });
 
-// Add call to decryption
-tx.moveCall({
-  target: `${SEAL_PACKAGE_ID}::bf_hmac_encryption::decrypt`,
+// 6. Construct the encrypted object.
```

Comment:
We already have the BCS-serialized encrypted object from earlier which we're passing here. Should we rephrase this instruction to differentiate from that somehow?